### ISSUE TITLE
Adding nuclear reactors

### ISSUE THREAD

**Issue by opain-replika** [forked-from-1kasper/loria#3]
Are there any plans of adding nuclear reactors to the game? I think it fits a lot the topic of the game that includes radioactive materials and whole radiation mechanic as well as electronic circuitry.

**Comment by rzrn** [comment]
Yes, there were thoughts on this idea. Currently there is RITEG in game, which is _not_ nuclear reactor, but it also converts nuclear energy into electrical (IRL first nuclear to thermal, then thermal to electrical, therefore RITEG has low efficiency). Unfortunately, there is now no craft for RITEG (someday I will fix it). If you want to try it, you can find it in creative mode (plutonium oxide can be used as fuel).
The problem is that the nuclear reactor is a very large system. I try to stay in development as close to the real world as possible, so I don’t want to implement reactor just a some “magic cube” or a collection of “magic cubes”, which only work if assembled in a certain way (how is it done in Industrial Craft 2). But how to do it better, I don’t know.
Unlike nuclear reactor, RITEG is quite simple, so its implementation just as one node is a little simplification.

**Comment by opain-replika** [comment]
Okay, so let me try to explain some things and maybe bring some ideas.
Nuclear reactor generators usually work kind of similarly as you described RITEG.
They transfer energy that's achieved by nuclear material reaching controlled state of reaching self sustained reaction to thermal energy, using which they in one way or another heat up water generating steam. Steam increases pressure and pushes turbine blades connected to generator and generate electrical power.
Then the steam has to condensate and be cooled in cooling towers and in form of cold water returned into reactor core to both cool it down to not allow it to melt but also to become the means of new steam to be generated.
This description is very simplified and it could be modified to be more IRL later, because there are other processes happening that are not implemented or may be hard to implement in loria.
Now if that reactor ought to be compact then there is no going around of making those "magical blocks", but building more authentic one might be a burden because it will be quite a large structure.

So yes, primitive reactor would need:

- fuel: for fuel most popular are such elements like plutonium and uranium;
- core activity accelerator: graphite and water (deuterium oxide)
- core activity decelerator: zirconium, lead and boron
- piping
- water pump
- turbine connected to generator
- water cooler: for initial implementation could be just a requirement to pass pipe into river or pond, might need temperature mechanics implementation
that's probably it for a primitive one (as reference used Russian RBMK design)

Need ability to:

- insert fuel into it
- insert control rods
- elevate/set down zirconium and/or lead rods to slow down or stop reaction
- elevate/set down graphite rods  to accelerate or slow down or put power to needed level
- turn on/off turbine gearbox
- switch gears on turbine gearbox (there are 2 one for startup sequence and another for normal operation)
- switch water pump on/off
and alike.
